File: nodejs/leap_year.js
const isLeapYear = (v) => {
  if (v % 4 != 0) return false;
  if (v % 100 == 0) return v % 400 == 0;
  return true;
};

const readline = require('readline').createInterface({
  input: process.stdin,
  output: process.stdout
});

readline.question('Please enter the year: ', (v) => {
  if (v == null || v == '' || Number.isNaN(Number(v))) {
    console.log(`'${v}' is not integer`);
    process.exit(1);
  }
  if (isLeapYear(Number(v))) {
    console.log(`'${v}' is leap year`);
  } else {
    console.log(`'${v}' is not leap year`);
  }
  readline.close();
});

File: python/leap_year.py
import sys

def is_leap_year(year: int):
    if year % 4 != 0:
        return False
    if year % 100 == 0:
        return year % 400 == 0
    return True

year = input('Please enter the year: ').strip()

try:
    num_year = int(year)
except ValueError as e:
    print(f'\'{year}\' is not integer')
    sys.exit(1)

if is_leap_year(num_year):
    print(f'\'{year}\' is leap year')
else:
    print(f'\'{year}\' is not leap year')
